server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -906,6 +924,52 @@ static Settings aggregateIndexSettings(
         return indexSettings;
     }
 
+    /**
+     * Updates index settings to enable remote store by default based on cluster level settings
+     * @param settingsBuilder index settings builder to be updated with relevant settings
+     * @param requestSettings settings passed in during index create request
+     * @param clusterSettings cluster level settings
+     */
+    private static void updateRemoteStoreSettings(Settings.Builder settingsBuilder, Settings requestSettings, Settings clusterSettings) {
+        if (CLUSTER_REMOTE_STORE_ENABLED_SETTING.get(clusterSettings)) {
+            // Verify if we can create a remote store based index based on user provided settings
+            if (!canCreateRemoteStoreIndex(requestSettings)) {
+                return;
+            }
+
+            settingsBuilder.put(SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT).put(SETTING_REMOTE_STORE_ENABLED, true);
```

Comment:
Why are we adding Segment Replication setting explicitly here? Are we sure that replication type will always be segment at this point? 
As I understand, Segment replication also has a cluster level setting. `settingsBuilder` should be updated by cluster and index level setting of segment replication prior to `updateRemoteStoreSettings`